CONTRIBUTING.md

```diff
@@ -0,0 +1,56 @@
+# Contributing
+
+## Tooling
+
+Not all of this is required to contribute, but for the best experience it's
+recommended you check all of this out.
+
+- [asdf](https://asdf-vm.com/)
+- [pnpm](https://pnpm.io/)
+- [typescript](https://www.typescriptlang.org/)
+- [prettier](https://prettier.io/)
+- [eslint](https://eslint.org/)
+- [turborepo](https://turbo.build/)
+- [cypress](https://www.cypress.io/)
+- [changesets](https://github.com/changesets/changesets#readme)
+
+## Getting started
+
+```
+git clone your-fork/mindgraph
+pnpm i
+pnpm dev
+```
+
+## Code style
+
+Feel free to do whatever you want for the most part! We have a few conventions that should be followed:
+
+```ts
+// "public" (exported) functions should be camel case
+export function doCoolStuff(): void;
+
+// "private" functions should be snake case
+function helper_function(): number;
+
+// classes should read public (top) to private (bottom)
+export class Cat {
+  public color: string;
+  constructor() {}
+  public runAround(): void;
+  private mood: number;
+  private speak_spanish(): string;
```

Comment:
I'm curious about this convention (the snake_case) - is this a standard convention, or just something specific to this javascript? I hadn't seen it before

Reply:
I figured it came from somewhere, thanks!